src/v/archival/ntp_archiver_service.cc

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "archival/ntp_archiver_service.h"
+
+#include "archival/logger.h"
+#include "archival/utils.h"
+#include "model/metadata.h"
+#include "s3/client.h"
+#include "s3/error.h"
+#include "storage/disk_log_impl.h"
+#include "storage/fs_utils.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include <fmt/format.h>
+
+#include <stdexcept>
+
+namespace archival {
+
+class arch_policy_iface {
+public:
+    virtual std::optional<manifest>
+    make_local_manifest(storage::log_manager& lm) = 0;
+};
+
+class arch_policy_non_compacted final : public arch_policy_iface {
```

Comment:
if you are making this an interface, why define the impl on the header file? testing?